File: discover_more_python/write_them_to_disc.py
# Function that gets githubs popular python repos and writes json to file.

from urllib2 import Request, urlopen, URLError

request_headers = {
                'User-Agent': 'Holberton School',
                'Authorization': 'token 170f8634262556331b3cb1809ffd4eb6e6fd40e1'
           }

request = Request('https://api.github.com/search/repositories?q=language:python&sort=stars&order=desc', headers=request_headers)

filename = '/tmp/35'
file = open(filename, 'w')
file.truncate()

try:
            response = urlopen(request)
            string = response.read()
            file.write(string)
            print "The file was saved!"
except URLError, e:
        print "Error code: ", e

        file.close()

File: discover_javascript/write_them_to_disc.js
const https = require('https');

var options = {
    hostname: 'api.github.com',
    path: '/search/repositories?q=language:javascript&sort=stars&order=desc',
    headers: {
        'User-Agent': 'Holberton_School',
        'Authorization': 'token 170f8634262556331b3cb1809ffd4eb6e6fd40e1'
    }
}

var req = https.request(options, function(res) {
    streamToString(res, function(jsonString) {
        var fs = require('fs');
        fs.writeFile('/tmp/35', jsonString, function (err) {
            if (err) return console.log(err);
            console.log('The file was saved!');
        });
    });
});
req.end();

function streamToString(stream, cb) {
    const chunks = [];
    stream.on('data', (chunk) => {
        chunks.push(chunk);
    });
    stream.on('end', () => {
        cb(chunks.join(''));
    });
}
